File: Examen de Lenguaje/C/BusquedaBinaria.c
#include <stdio.h>
#include <windows.h>

int main() {
    int arreglo[100];
    int n, i, objetivo, izquierda, derecha, medio;
    int encontrado = 0;

    printf("Ingrese el numero de elementos del arreglo: ");
    scanf("%d", &n);

    printf("Ingrese los elementos del arreglo:\n");
    for (i = 0; i < n; i++) {
        scanf("%d", &arreglo[i]);
    }

    printf("Ingrese el numero a buscar: ");
    scanf("%d", &objetivo);

    izquierda = 0;
    derecha = n - 1;

    while (izquierda <= derecha && !encontrado) {
        medio = (izquierda + derecha) / 2;
        if (arreglo[medio] == objetivo) {
            printf("Numero encontrado en la posicion: %d\n", medio + 1);
            encontrado = 1;
        } else if (arreglo[medio] < objetivo) {
            izquierda = medio + 1;
        } else {
            derecha = medio - 1;
        }
    }

    if (!encontrado)
        printf("Numero no encontrado.\n");

    printf("Presione ENTER para salir...");
    getchar(); // limpiar buffer
    getchar(); // esperar ENTER
    return 0;
}

File: Examen de Lenguaje/Fortran/BusquedaBinaria.f90
program BusquedaBinaria
    implicit none
    integer, dimension(100) :: arreglo
    integer :: n, i, objetivo, izquierda, derecha, medio
    logical :: encontrado
    character(len=1) :: pausa

    print *, "Ingrese el numero de elementos del arreglo:"
    read(*,*) n

    print *, "Ingrese los elementos del arreglo:"
    do i = 1, n
        read(*,*) arreglo(i)
    end do

    print *, "Ingrese el numero a buscar:"
    read(*,*) objetivo

    izquierda = 1
    derecha = n
    encontrado = .false.

    do while (izquierda <= derecha .and. .not. encontrado)
        medio = (izquierda + derecha) / 2
        if (arreglo(medio) == objetivo) then
            print *, "Numero encontrado en la posicion:", medio
            encontrado = .true.
        else if (arreglo(medio) < objetivo) then
            izquierda = medio + 1
        else
            derecha = medio - 1
        end if
    end do

    if (.not. encontrado) print *, "Numero no encontrado."

    print *, "Presione ENTER para salir..."
    read(*,*) pausa
end program BusquedaBinaria

